### PR TITLE
Treat empty files in pinning as if they weren't available

### PR DESCRIPTION
**Pull request by gergelyfabian**
We have a strange behavior that in our artifactory server source jars happen to sometimes appear as a 0-byte jar.
We'd like to treat them as if they weren't there.

There is a legitimate question here though, is it legal to have a 0-byte jar file? If this is a legal case, then I'd be happy to change this PR to hide such a feature behind a switch (to be able to remove such jars from the results of coursier), otherwise will leave this on a fork.

### CODE REVIEW

**Comment by gergelyfabian** [comment]
> There is a legitimate question here though, is it legal to have a 0-byte jar file? If this is a legal case, then I'd be happy to change this PR to hide such a feature behind a switch (to be able to remove such jars from the results of coursier), otherwise will leave this on a fork.

I've hidden the feature behind a feature flag.

File: private/extensions/maven.bzl [hunk 1]
```diff
@@ -170,6 +171,7 @@ def _maven_impl(mctx):
     # - strict_visibility_value: a string list. Build will fail is duplicated and different.
     # - use_starlark_android_rules: bool. A logical OR over all `use_starlark_android_rules` for all `install` tags with the same name.
     # - version_conflict_policy: string. Fails build if different and not a default.
+    # - ignore_empty_files: Treat jars that are empty as if they were not found.
```
Comment (cheister) [inline private/extensions/maven.bzl:174]: Do you mind adding something to the README.md for this option?

**Comment by cheister** [comment]
Thanks for the PR. 

I'm sort of surprised that maven repositories will serve 0 byte jars.  Have you only seen this with source jars or with non-source jars as well?

File: private/extensions/maven.bzl (hunk 1 above)
Comment (gergelyfabian) [reply private/extensions/maven.bzl:174]: Sure, I could do that.

**Comment by gergelyfabian** [comment]
> Thanks for the PR.
> 
> I'm sort of surprised that maven repositories will serve 0 byte jars. Have you only seen this with source jars or with non-source jars as well?

Yes, it's unknown why these empty jars are appearing. We have an artifactory where this is appearing, when downloading artifacts from a 3rd party artifactory. It's only appearing for source jars.

File: private/extensions/maven.bzl (hunk 1 above)
Comment (gergelyfabian) [reply private/extensions/maven.bzl:174]: Done.

**Review by cheister — APPROVED** [review]

**Comment by gergelyfabian** [comment]
Resolved conflicts and fixed one bug.

**Review by shs96c — APPROVED** [review]
LGTM. Thank you for your patience with the review!

**Comment by gergelyfabian** [comment]
I'm not sure about the test failures... They fail e.g. due to:

```
Error: 'JavaInfo' value has no field or method 'transitive_runtime_deps'
...
Error: 'JavaInfo' value has no field or method 'transitive_deps'
...
error: Source option 7 is no longer supported. Use 8 or later.
error: Target option 7 is no longer supported. Use 8 or later.
```

I think this is not related to this PR. In fact master is also failing.

**Comment by shs96c** [comment]
The build appears to be failing for unrelated issues (`JavaInfo` no longer having `transitive_runtime_deps`), so merging.

**Comment by gergelyfabian** [comment]
> The build appears to be failing for unrelated issues (`JavaInfo` no longer having `transitive_runtime_deps`), so merging.

Rebased with master after the build failures were fixed there.

**Pull request merged**